File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def minDepth(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if root is None:
            return 0
        def recurse(node):
            if node is None:
                return float('inf')
            if node.left is None and node.right is None:
                return 1
            left_depth = recurse(node.left)
            right_depth = recurse(node.right)
            return min(left_depth, right_depth) + 1

        return recurse(root)

File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left),
 * right(right) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode* root) {
        if (root == nullptr)
            return 0;
        queue<TreeNode*> nodeque;
        nodeque.push(root);
        int depth = 0;
        while (!nodeque.empty()) {
            int size = nodeque.size();
            depth++;
            for (int i = 0; i < size; i++) {
                TreeNode* cur = nodeque.front();
                nodeque.pop();
                if (cur->left)
                    nodeque.push(cur->left);
                if (cur->right)
                    nodeque.push(cur->right);
                if (!cur->left && !cur->right)
                    return depth;
            }
        }
        return depth;
    }
};